### PR TITLE
feat(linter): add missing test cases to no-empty-interface and add config

### PR DESCRIPTION
**Pull request by JoSeBu1**
Based on this https://github.com/oxc-project/oxc/issues/2934 I added missing test cases and rule configuration

### CODE REVIEW

**Review by Boshen — APPROVED** [review]

File: crates/oxc_linter/src/rules/typescript/no_empty_interface.rs [hunk 1]
```diff
@@ -66,25 +79,134 @@ fn test() {
...
-        "interface Foo { props: string; } interface Bar extends Foo {} class Bar {}",
-        "interface Foo { props: string; } interface Bar extends Foo {} const bar = class Bar {};",
-        "interface Foo { name: string; } interface Bar extends Foo {}",
-        "interface Foo extends Array<number> {}",
-        "interface Foo extends Array<number | {}> {}",
-        "interface Bar { bar: string; } interface Foo extends Array<Bar> {}",
-        "type R = Record<string, unknown>; interface Foo extends R {}",
-        "interface Foo<T> extends Bar<T> {}",
-        "declare module FooBar { type Baz = typeof baz; export interface Bar extends Baz {} }",
+        ("interface Foo {}", None),
```
Comment (mysteryven) [inline crates/oxc_linter/src/rules/typescript/no_empty_interface.rs:130]: Looks also need to add [`interface Foo extends {}`](https://github.com/typescript-eslint/typescript-eslint/blob/a6ab2cb9be5a09498e953896e7fad04b2c53ea42/packages/eslint-plugin/tests/rules/no-empty-interface.test.ts#L64)

Comment (JoSeBu1) [reply crates/oxc_linter/src/rules/typescript/no_empty_interface.rs:130]: mmm, this was not added by codegen
